File: Day Six/Java phasegate/AgeCalculator.java
import java.util.Scanner;
import java.time.LocalDate;
	
	public class AgeCalculator { 

		public static int calculateAge(String DOB) { 
		String characterRemoval = DOB.replaceAll("\\D", "");
		int dateOfBirth = Integer.parseInt(characterRemoval);
		int day = dateOfBirth / 1000000;
		int month = (dateOfBirth / 10000) % 100;
		int year = dateOfBirth % 10000;
		
		LocalDate birthdate = LocalDate.of(year, month, day);
		LocalDate currentDOB = LocalDate.now();
		int presentAge = currentDOB.getYear() - birthdate.getYear();

		if (currentDOB.getMonthValue() < month ||
           	 (currentDOB.getMonthValue() == month &&
            	 currentDOB.getDayOfMonth() < day)) {
           		presentAge--;
       		 }

        return presentAge;
    }


public static void main(String[] args) { 
Scanner scanner = new Scanner(System.in); 
System.out.print("Enter date of birth: ");
String dateOfBirth = scanner.nextLine();
System.out.print(calculateAge(dateOfBirth));

		
	}
}

File: Day Six/JavaScript phasegate/AgeCalculator.js
function calculateAge(String DOB) { 
		const characterRemoval = DOB.replace(/\D/g, "");
		const dateOfBirth = parseInt(characterRemoval);
		const day = dateOfBirth // 1000000;
		const month = (dateOfBirth // 10000) % 100;
		const year = dateOfBirth % 10000;
		
		const birthdate = LocalDate.of(year, month, day);
		const currentDOB = new Date();
		let presentAge = currentDOB.getYear() - birthdate.getYear();
		

		if (currentDOB.getMonth() < birthdate.getMonth ||
           	 (currentDOB.getMonth() == birthdate.getMonth &&
            	 currentDOB.getDate() < birthday.getDate)) {
           		presentAge--;
       		 }

        return presentAge;
	}    
dateOfBirth = prompt("Enter date of birth: ");
console.log(calculateAge(dateOfBirth));
